src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,94 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75F;
+    private Node<K, V>[] nodes;
+    private int size;
+    private double threshold;
+
+    public MyHashMap() {
+        nodes = (Node<K, V>[]) new Node[DEFAULT_CAPACITY];
+        threshold = LOAD_FACTOR * DEFAULT_CAPACITY;
+        size = 0;
+    }
+
+    private class Node<K, V> {
+        private Node<K, V> next;
+        private K key;
+        private V value;
+        private int hash;
+
+        private Node(Node<K, V> next, K key, V value, int hash) {
+            this.next = next;
+            this.key = key;
+            this.value = value;
+            this.hash = hash;
+        }
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (size + 1 > threshold) {
+            resize();
+        }
+        int index = getHash(key);
+        Node<K, V> node = nodes[index];
+        if (node == null) {
+            nodes[index] = new Node<>(null, key, value, index);
+            size++;
```

Comment:
we can make `size++` after `if-else` block at one place, or we can add `return` here and remove `else` block